aqua/qt6/Portfile

```diff
@@ -1313,17 +1313,6 @@ post-destroot {
                 ln -s ${qt_bins_dir}/${cmakefile} ${destroot}${mp_prefix}/lib/cmake/${cmakefile}
             }
         }
-
-        # if cmake finds configuration files in ${mp_prefix}/lib/cmake, CMAKE_CURRENT_LIST_DIR expands to ${mp_prefix}/lib/cmake/xxx
-        # cmake configuration files actually installed in ${qt_cmake_module_dir}/xxx
-        set cmakepath ${destroot}${qt_libs_dir}/cmake/
-        foreach cmakefile [ rglob ${cmakepath} *.cmake ] {
-            if { ![file isdirectory ${cmakefile} ] } {
-                set cmakedir [file dirname ${cmakefile}]
-                set cmakedir [string replace $cmakedir 0 [string length $cmakepath]-1]
-                reinplace -q "s|\\\${CMAKE_CURRENT_LIST_DIR}|${qt_cmake_module_dir}/${cmakedir}|g" ${cmakefile}
-            }
-        }
```

Comment:
Getting the following failure trying to configure a dolphin build:
```
CMake Error at /opt/local/lib/cmake/Qt6CoreTools/Qt6CoreToolsTargets.cmake:154 (message):
  The imported target "Qt6::moc" references the file

     "/opt/local/bin/moc"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/opt/local/lib/cmake/Qt6CoreTools/Qt6CoreToolsTargets.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  /opt/local/lib/cmake/Qt6CoreTools/Qt6CoreToolsConfig.cmake:37 (include)
  /opt/local/lib/cmake/Qt6Core/Qt6CoreDependencies.cmake:76 (find_package)
  /opt/local/lib/cmake/Qt6Core/Qt6CoreConfig.cmake:40 (include)
  /opt/local/lib/cmake/Qt6/Qt6Config.cmake:209 (find_package)
  Source/Core/DolphinQt/CMakeLists.txt:22 (find_package)
```
This looks awfully similar to what this code claims to fix...
(Putting this code back fixes it)

Reply:
How do you configure? With qmake or qt-cmake ?

You may see welle.io portfile for an example that uses qt-cmake (welle.io-devel subport actually) towards qt6

Reply:
Restoring these lines is not the way to go IMO.

Reply:
See also #17071

Reply:
With normal `cmake`, with the `CMakeLists.txt` searching for qt with `find_package(Qt6 COMPONENTS Core Gui Widgets)`

Reply:
Note: This is not a port.  This is just me attempting to use macports' qt6 to satisfy a dependency when building a project from source.  AKA option 3 here:
https://github.com/macports/macports-ports/blob/223bfc67382a039df400d0b9323e75af984c3ed8/aqua/qt6/Portfile#L624

IMO this should continue to work.  I've posted an alternative way of making this work on #17071, if you'd prefer that.

Reply:
As per qt6 documentation: 
https://doc.qt.io/qt-6/cmake-build-on-cmdline.html